src/main/java/com/puppycrawl/tools/checkstyle/doclets/TokenTypesDoclet.java

```diff
@@ -19,148 +19,237 @@
- * (first sentences) of TokenTypes' constants.
+ * This class is used internally in the build process to write a property file
+ * with short descriptions (the first sentences) of TokenTypes constants.
  * Request: 724871
- * For ide plugins (like the eclipse plugin) it would be useful to have
- * programmatic access to the first sentence of the TokenType constants,
+ * For IDE plugins (like the eclipse plugin) it would be useful to have
+ * a programmatic access to the first sentence of the TokenType constants,
  * so they can use them in their configuration gui.
- * @author o_sukhodolsky
+ * @author Pavel Bludov
+ * @noinspection UseOfSystemOutOrSystemErr
  */
 public final class TokenTypesDoclet {
-    /** Command line option to specify file to write output of the doclet. */
-    private static final String DEST_FILE_OPT = "-destfile";
 
-    /** Stop instances being created. */
+    /**
+     * The command line option to specify the output file.
+     */
+    private static final String OPTION_DEST_FILE = "destfile";
+
+    /**
+     * The width of the CLI help option.
+     */
+    private static final int HELP_WIDTH = 100;
+
+    /**
+     * This regexp is used to convert a multiline javadoc to a single line without asterisks.
+     */
+    private static final Pattern JAVADOC_MULTILINE_TO_SINGLELINE_PATTERN =
+        Pattern.compile("(\\r?)\\n\\s*\\*", Pattern.DOTALL);
+
+    /**
+     * This regexp is used to extract the first sentence from the text.
+     */
+    private static final Pattern FIRST_SENTENCE_PATTERN = Pattern.compile("\\s*(.*?)\\.(\\s|$)");
+
+    /**
+     * This regexp is used to convert inline tags to HTML tags.
+     */
+    private static final Pattern INLINE_TAG_PATTERN = Pattern.compile("\\{@(\\w+)\\s+(.*?)}");
+
+    /**
+     * Don't create instance of this class, use the {@link #main(String[])} method instead.
+     */
     private TokenTypesDoclet() {
     }
 
     /**
-     * The doclet's starter method.
-     * @param root {@code RootDoc} given to the doclet
-     * @return true if the given {@code RootDoc} is processed.
-     * @exception FileNotFoundException will be thrown if the doclet
-     *            will be unable to write to the specified file.
+     * TokenTypes.properties generator entry point.
+     * @param args the command line arguments.
+     * @throws IOException if there is a problem with files access
+     * @noinspection CallToPrintStackTrace
+     **/
+    public static void main(String... args) throws IOException {
+        try {
+            final CommandLine commandLine = parseCli(args);
+            if (commandLine.getArgList().size() == 1) {
+                writePropertiesFile(commandLine.getArgList().get(0),
+                    commandLine.getOptionValue(OPTION_DEST_FILE));
+            }
+            else {
+                printUsage();
+            }
+        }
+        catch (ParseException ex) {
+            System.out.println(ex.getMessage());
```

Comment:
Shouldn't the exception be passed all the way up?
We need a build failure to happen if an exception occurs. No one will be looking at the console for a stack trace unless there is a build failure.